### PR TITLE
Use Java mechanisms to read language files and default to UTF-8

### LINKED ISSUE #21753: Standardize message property handling for translation tools
### Description

As of JDK9, Java support Java property resource bundles to be encoded in UTF-8. At the moment Keycloak uses a non-standard mechanism for this which is difficult to handle by standard translation tools.

This uses Java's standard mechanisms

### Discussion

https://github.com/keycloak/keycloak/discussions/9270#discussioncomment-6468612

### Motivation

Better support for standard localization tools which don't read now write Keycloak's comments on the first line.

### Details

I'll prepare a PR for this.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #21753

Discussion: https://github.com/keycloak/keycloak/discussions/9270

### CODE REVIEW

**Comment by ahus1** [comment]
While this is still in draft, it already contains updates to all property files and the docs.

**Review by jonkoops — DISMISSED** [review]

**Comment by ahus1** [comment]
It would be great to have a review from one of the maintainers on the Java side on this. Thanks!

**Review by hmlnarik — COMMENTED** [review]
Approving for the Java part

**Review by hmlnarik — DISMISSED** [review]

**Review by ghost — COMMENTED** [review]
Unreported flaky test detected, please review

**Comment by ahus1** [comment]
@jonkoops - I've look at the CodeQL alert, and also added a Regex check for filenames - CodeQL still complains. The logic to create the filename didn't change in this PR, so I assume it is as safe as before. I'll follow up with Bruno to find out why CodeQL complains here.

**Comment by jonkoops** [comment]
@ahus1 sounds good to me, should we go ahead and merge this then?

**Comment by ahus1** [comment]
@jonkoops - yes, I think this is the way to go. Please approve it for the UI part. 
I'll also get an re-approval of a maintainer/Java developer, and then it will be ready to be merged.

**Review by jonkoops — DISMISSED** [review]
LGTM

**Comment by ahus1** [comment]
@hmlnarik - I've added one more commit to make CodeQL happy, but it continues to warn. I'd like to keep that change, and have it squashed when committing. I'll contact @abstractj separately to figure out how to make CodeQL happy.

Could you please re-approve and merge? Thanks!

**Comment by ahus1** [comment]
Rebased to resolve conflicts on the docs. Also squashed the two commits.

File: docs/documentation/release_notes/topics/23_0_0.adoc
Comment (mhajas) [inline docs/documentation/release_notes/topics/23_0_0.adoc:3]:
```suggestion
Message properties files for themes are now read in UTF-8 encoding, with an automatic fallback to ISO-8859-1 encoding.
```

File: docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc
Comment (mhajas) [inline docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc:21]: Is it possible that someone uses some property files encoded in a different encoding than `UTF-8` or `ISO-8859-1`?

**Review by mhajas — CHANGES_REQUESTED** [review]
thank you for the PR @ahus1, I added two comments. Also, it seems a rebase is needed (sorry for the late review).

File: docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc
Comment (ahus1) [reply docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc]: Yes, it is possible that people used different encodings in the past, but no longer supported. 

I've added one more sentence to the migration guide: 

> If you are using a different encoding, convert the files to UTF-8.

**Review by mhajas — APPROVED** [review]
Thank you @ahus1

**Review by jonkoops — APPROVED** [review]

**Pull request merged**
